File: app/src/main/java/ua/com/zak/budgetswing/BudgetApplication.java
package ua.com.zak.budgetswing;

import android.app.Application;

import ua.com.zak.budgetswing.core.dao.factory.SimpleDaoFactory;
import ua.com.zak.budgetswing.core.di.ApplicationGraph;
import ua.com.zak.budgetswing.navigator.AndroidNavigator;

/**
 * @author zak <zak@swingpulse.com>
 */
public class BudgetApplication extends Application {

    private ApplicationGraph mApplicationGraph;

    @Override
    public void onCreate() {
        super.onCreate();
        mApplicationGraph = new ApplicationGraph(
                new SimpleDaoFactory(),
                new AndroidNavigator()
        );
    }

    public ApplicationGraph getApplicationGraph() {
        return mApplicationGraph;
    }
}

File: app/src/main/java/com/robotnec/budget/refactored/BudgetApplication.kt
package com.robotnec.budget.refactored

import androidx.appcompat.app.AppCompatDelegate
import com.jakewharton.threetenabp.AndroidThreeTen
import com.robotnec.budget.refactored.di.DaggerApplicationComponent
import dagger.android.support.DaggerApplication

/**
 * @author zak zak@swingpulse.com>
 */
class BudgetApplication : DaggerApplication() {

    public override fun applicationInjector() = DaggerApplicationComponent.builder()
            .application(this)
            .build()

    override fun onCreate() {
        super.onCreate()
        AndroidThreeTen.init(this)
        AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_YES)
    }
}
